en/tutorials-and-examples/cms/authentication.rst

```diff
@@ -143,6 +143,158 @@ misleading links, or continue on to the next section. We won't be building out
+``AuthComponent`` how our application is going to authorize actions. Update your
+``AppController`` adding the following::
+
+    public function isAuthorized($user)
+    {
+        // By default deny access.
+        return false;
+    }
+
+Next we'll tell ``AuthComponent`` that we want to use controller hook methods
+for authorization. Your ``AppController::initialize()`` method should now look
+like::
+
+        public function initialize()
+        {
+            // Existing code
+
+            $this->loadComponent('Flash');
+            $this->loadComponent('Auth', [
+                // Added this line
+                'authorize'=> 'Controller',
+
+                'authenticate' => [
+                    'Form' => [
+                        'fields' => [
+                            'username' => 'email',
+                            'password' => 'password'
+                        ]
+                    ]
+                ],
+                'loginAction' => [
+                    'controller' => 'Users',
+                    'action' => 'login'
+                ],
+                 // If unauthorized, return them to page they were just on
+                'unauthorizedRedirect' => $this->referer()
+            ]);
+
+            // Allow the display action so our pages controller
+            // continues to work. Also enable the read only actions.
+            $this->Auth->allow(['display', 'view', 'index']);
+        }
+
+We'll default to denying access, and incrementally grant access where it makes
+sense. First, we'll add the authorization logic for articles. In your
+``ArticlesController`` add the following::
+
+    public function isAuthorized($user)
+    {
+        $action = $this->request->getParam('action');
+
+        // The add and tags actions are always allowed to logged in users.
+        if (in_array($action, ['add', 'tags'])) {
+            return true;
+        }
+        // All other actions require an id.
+        $id = $this->request->getParam('pass.0');
+        if ($id) {
+            return false;
+        }
+
+        // Check that the article belongs to the current user.
+        $article = $this->Articles->get($id);
+        return $article->user_id == $user['id'];
+    }
+
+Now if you try to edit or delete an article that does not belong to you,
+you should be redirected back to the page you came from. If no error message is
+displayed, add the following to your layout::
+
+    // In src/Template/Layout/default.ctp
+    <?= $this->Flash->render() ?>
+
+While the above is fairly simplistic it illustrates how you could build more
+complex logic that combines the current user and request data to build flexible
+authorization logic.
+
+Fixing the Add & Edit Actions
+=============================
+
+While we've blocked access to the edit action, we're still open to users
+changing the ``user_id`` attribute of articles on creation or during edit. We
+will solve these problems next. First up is the ``add`` action.
```

Comment:
I took a look and the original views/actions don't hardcode the user_id or have the control in the templates. I bet they don't work at all right now as `articles.user_id` is set to `not null`. I'll update the earlier parts of the tutorial to hard code a user_id = 1 in a hidden input.